main.py

```diff
@@ -4,16 +4,25 @@
 from typing import Mapping, NamedTuple, Tuple
 from uuid import UUID
 
+import google.auth
 import jwt
 from flask import Request
+from google.cloud import secretmanager
 from requests import Session
 from requests.exceptions import RequestException
 
 from exceptions import InvalidNotifyKeyError, InvalidRequestError
 
-NOTIFY_API_KEY = os.environ["NOTIFY_API_KEY"]
 NOTIFY_BASE_URL = "https://api.notifications.service.gov.uk/v2"
 
+if (NOTIFY_API_KEY := os.getenv("NOTIFY_API_KEY")) is None:
+    _, project_id = google.auth.default()
+    client = secretmanager.SecretManagerServiceClient()
+    access_secret_version_response = client.access_secret_version(
```

Comment:
I think naming this what it is rather than directly after the method name it's calling would be clearer, something like `secret_manager_response`.

Reply:
Its the name of the response object that is returned from that method (AccessSecretVersionResponse), but will update to secret_manager_response